contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,125 @@
+# Priority in ResourceQuota
+
+Authors:
+
+@resouer @bsalamat @derekwaynecarr 
+
+Sep 2017
+
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota](#resourcequota)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, 3 sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies priority class names.
```

Comment:
this is the part that has been bugging me forever as i think it represents a deficiency with quota.  in theory, any other object type may want to scope quota further, and we cannot really support a system where we can add arbitrary fields like this for each scenario.  maybe pods are special because they are in the core, but its this part of this proposal that has hung me up for awhile as its not a feature we would extend to other aggregated api servers.

when we first added scopes, i attempted to implement it using a field selector, but we had no official support for field selectors in the project that could be used in types.  i still feel like this calls out another example where field selectors would have been useful.